src/decrypt-body.js

```diff
@@ -0,0 +1,52 @@
+/**
+
+Copyright 2018 New Vector Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+**/
+
+const { PkDecryption } = require('olm');
+const ClientError = require('../src/client-error.js');
+
+const decryption = new PkDecryption();
+const publicKey = decryption.generate_key();
```

Comment:
Is the intention to read this from disk at some point?

Reply:
Hm. We could make it a config option so that the client only needs to get the public key once instead of every time it needs to do a request.

I imagined the public key is stored as state in the decryption logic but this is not true. The `generate_key` method actually just returns a random string.

Reply:
>@uhohreg:matrix.org: Yes, I'm planning on making a pickle/unpickle method for persisting and loading the decryption key.  I just haven't managed to get that done yet.  Hopefully I'll get that done tomorrow.  For now, just generate a new key every time the server starts.

Reply:
ah, cool